Task №10

jacsmedvedev opened the pull request:
This is first commit of this task. The discription of the task in
Pasport class.

jacsmedvedev committed:
Task №10

This is first commit of this task. The discription of the task in
Pasport class/.

strazhkoe commented on src/task_from_teacher_10/InternationalPassport.java:
```diff
@@ -0,0 +1,49 @@
+
+public class InternationalPassport extends Pasport {
+		
+	private Visa _visa;
```
Загранпаспорт должен предусматривать наличие нескольких виз. Переделай с использованием коллекций

strazhkoe commented on src/task_from_teacher_10/Main.java:
```diff
@@ -0,0 +1,69 @@
+ * In this class we declared and initialized object InternationalPassport.
+ * 
+ * @author Yakov Medvedev
+ */
```
Этот комментарий должен быть сразу над классом, а не над пакетом

strazhkoe commented on src/task_from_teacher_10/Visa.java:
```diff
@@ -0,0 +1,49 @@
+	public Visa( final VisaType visaType, 
+				 final LocalDate openingDate, 
+				 final LocalDate closingDate, 
+				 final Sex sex) {
```
Зачем в визе пол? А где страна?

jacsmedvedev committed:
Task №10 commit#2

Исправил согласно замечаниям:
1.Переделал с использованием коллекций 
(теперь можно добавлять визы);
2.Перенес комментарии;
3.Убрал поле "пол" и добавил поле "Страна" в визе.

strazhkoe commented on src/task_from_teacher_10/InternationalPassport.java:
```diff
@@ -0,0 +1,56 @@
+			int yearOfIssue,
+			int mounthOfIssue, 
+			int dayOfIssu) {
+		super(country, pasportSeries, pasportNumber, surname, name, patronymic, 
```
Слишком много параметров. Давай часть, которая относится к обычному паспорту, будем передавать в объекте типа паспорт сюда

strazhkoe commented on src/task_from_teacher_10/InternationalPassport.java:
```diff
@@ -0,0 +1,56 @@
+	public ArrayList <Visa> getVisa() {
+		return _visa;
+	}
+	public void setVisa(VisaType visaType, ForeignCountries foreignCountries, 
```
foreignCountries - неправильное имя поля, так как речь тут идет об одной стране

strazhkoe commented on src/task_from_teacher_10/InternationalPassport.java:
```diff
@@ -0,0 +1,56 @@
+	}
+
+	public void setInternationalPassportID(Integer internationalPassportID) {
+		_internationalPassportID = internationalPassportID;
```
тут надо проверять на null, так как передается Integer. Мы же не хотим чтоб паспорт остался без номера?

strazhkoe commented on src/task_from_teacher_10/Main.java:
```diff
@@ -0,0 +1,61 @@
+	interPasToString(interPas);
+	}
+
+	private static void interPasToString(InternationalPassport interPas) {
```
Это надо перенести в метод toString паспорта

jacsmedvedev committed:
Task №10 commit №2

1.Убрал некоторые параметры из конструктора InternationalPassport() и 
вставил Pasport ukrainePasport;
2.Переименовал foreignCountries на DepartureCountry;
3.В классе InternationalPassport добавил проверку на  null 
для _internationalPassportID;
4. toString() перенес и з Main в классы Passport() и
InternationalPassport().
